galera/src/replicator_smm.cpp

```diff
@@ -1470,9 +1470,30 @@ galera::ReplicatorSMM::process_conf_change(void*                    recv_ctx,
         if (st_required && app_wants_st)
         {
             // GCache::Seqno_reset() happens here
-            request_state_transfer (recv_ctx,
-                                    group_uuid, group_seqno, app_req,
-                                    app_req_len);
+            long ret =
+                request_state_transfer (recv_ctx,
+                                        group_uuid, group_seqno, app_req,
+                                        app_req_len);
+
+            if (ret < 0 || sst_state_ == SST_CANCELED)
+            {
+                // If the request was canceled due to an error at
+                // the GCS level or if IST/SST request was canceled
+                // by another thread (by initiative of the server),
+                // and if the node remain in the S_JOINING state,
+                // then we must return it to the S_CONNECTED state.
+                // The S_CONNECTED is the original state, which
+                // exist before the request_state_transfer started.
+                // If state transfer failed then we move the node
+                // back to old state. S_CONNECTED state will help
+                // us restart SST, especially if mysqldump method
+                // will used for state transfer:
```

Comment:
So are we saying we can restart SST only if mysqldump method is used ?
Even rsync or xtrabackup should able to restart it as they cleanup the directory ?